readme.md

```diff
@@ -131,6 +131,46 @@ const store = new Conf({
 
 > Note: The version the migrations use refers to the **project version** by default. If you want to change this behavior, specify the [`projectVersion`](#projectVersion) option.
 
+### beforeMigration
+
+Type: `Function`\
+Default: `undefined`
+
+You can use `beforeMigration` callback for logging purposes, preparing migration data, etc.
+
+`beforeMigration` callback will be called before the migration is executed.
+
+It could be useful when u have multiple configs and you want to log the migration data for each config.
+
+Example:
+
+```js
+const Conf = require('conf');
+
+console.log = someLogger.log;
+
+const mainConfig = new Conf({
+	beforeMigration: (store, currentVersion, nextVersion) => {
```

Comment:
I would prefer if it were like this.
```suggestion
	beforeMigration: (store, context) => {
```

That way we could easily add more metadata properties in the future.

Reply:
Should the properties maybe be named `fromVersion` and `toVersion`? `currentVersion` makes it sound like it's the current version of the package, not the current version in the migration process.

Reply:
> We could maybe also add the package version and all the migration versions. In case the user wants to know where in the migration process they are.


Added `context.migrationProcess` with `finalVersion` and `migrations` not sure how to name them properly probably they can be flattened and lay in context but proper naming required

Reply:
I would prefer to keep them top-level in `context` (no nesting).